packages/colorpickers/src/elements/ColorPicker/reducer.ts

```diff
@@ -18,12 +19,57 @@ type ColorPickerActionTypes =
   | { type: 'RED_CHANGE'; payload: string }
   | { type: 'GREEN_CHANGE'; payload: string }
   | { type: 'BLUE_CHANGE'; payload: string }
-  | { type: 'ALPHA_CHANGE'; payload: string };
+  | { type: 'ALPHA_CHANGE'; payload: string }
+  | { type: 'RESET_COLOR'; payload: IColor };
 
-type ReducerType = (state: IColor, action: ColorPickerActionTypes) => IColor;
+interface IColorPickerState {
+  color: IColor;
+  hexInput: string;
+  redInput: string;
+  blueInput: string;
+  greenInput: string;
+  alphaInput: string;
+}
+
+export function convertStringToColor(colorString: string): IColor | undefined {
```

Comment:
Extracted from `getInitialState` as this is used in many areas. Could probably be moved to the color conversion utility, but left it here for this review.